app/models/event_stream.rb

```diff
@@ -3,6 +3,7 @@ class EventStream < ApplicationRecord
 
   belongs_to :target, :polymorphic => true
   belongs_to :ext_management_system, :foreign_key => :ems_id
+  belongs_to :source_ext, :class_name => "ExtManagementSystem", :foreign_key => :source_ems_id
```

Comment:
`source_ems` would be a more common name for the association.
You can also drop the `foreign_key` option then